docs/pages/pmd/userdocs/migrating_to_pmd7.md

```diff
@@ -0,0 +1,2081 @@
+```
+
+See {% jdoc core::lang.ast.NodeStream %} for the details.
+Note: This was implemented via [PR #1622 [core] NodeStream API](https://github.com/pmd/pmd/pull/1622)
+
+### XPath: Migrating from 1.0 to 2.0
+
+XPath 1.0 and 2.0 have some incompatibilities. The [XPath 2.0 specification](https://www.w3.org/TR/xpath20/#id-incompat-in-false-mode)
+describes them precisely. Those are however mostly corner cases and XPath
+rules usually don't feature any of them.
+
+The incompatibilities that are most relevant to migrating your rules are not
+caused by the specification, but by the different engines we use to run
+XPath 1.0 and 2.0 queries. Here's a list of known incompatibilities:
+
+* The namespace prefixes `fn:` and `string:` should not be mentioned explicitly.
+  In XPath 2.0 mode, the engine will complain about an undeclared namespace, but
+  the functions are in the default namespace. Removing the namespace prefixes fixes it.
+  * <code><b style="color:red">fn:</b>substring("Foo", 1)</code> &rarr; `substring("Foo", 1)`
+* Conversely, calls to custom PMD functions like `typeIs` *must* be prefixed
+  with the namespace of the declaring module (`pmd-java`).
+  * `typeIs("Foo")` &rarr; <code><b style="color:green">pmd-java:</b>typeIs("Foo")</code>
+* Boolean attribute values on our 1.0 engine are represented as the string values
+  `"true"` and `"false"`. In 2.0 mode though, boolean values are truly represented
+  as boolean values, which in XPath may only be obtained through the functions
+  `true()` and `false()`.
+  If your XPath 1.0 rule tests an attribute like `@Private="true"`, then it just
+  needs to be changed to `@Private=true()` when migrating. A type error will warn
+  you that you must update the comparison. More is explained on [issue #1244](https://github.com/pmd/pmd/issues/1244).
+  * `"true"`, `'true'` &rarr; `true()`
+  * `"false"`, `'false'` &rarr; `false()`
+
+* In XPath 1.0, comparing a number to a string coerces the string to a number.
+  In XPath 2.0, a type error occurs. Like for boolean values, numeric values are
+  represented by our 1.0 implementation as strings, meaning that `@BeginLine > "1"`
+  worked ---that's not the case in 2.0 mode.
+  * <code>@ArgumentCount > <b style="color:red">'</b>1<b style="color:red">'</b></code> &rarr; `@ArgumentCount > 1`
+
+* In XPath 1.0, the expression `/Foo` matches the *children* of the root named `Foo`.
+  In XPath 2.0, that expression matches the root, if it is named `Foo`. Consider the following tree:
+  ```java
+  Foo
+  └─ Foo
+  └─ Foo
+  ```
+  Then `/Foo` will match the root in XPath 2.0, and the other nodes (but not the root) in XPath 1.0.
+  See e.g. [an issue caused by this](https://github.com/pmd/pmd/issues/1919#issuecomment-512865434) in Apex,
+  with nested classes.
+
+* The custom function "pmd:matches" has been removed, since there is a built-in function available since XPath 2.0
+  which can be used instead.
```

Comment:
I would be explicit as to what that function is (`matches`, with no prefix) and arguments (I believe it's identical, so just removing the prefix should suffice)